src/Validation/URLValidationRESTController.php

```diff
@@ -240,6 +313,32 @@ public function validate_post_url( $request ) {
 		return rest_ensure_response( $this->filter_response_by_context( $data, $request['context'] ) );
 	}
 
+	/**
+	 * Returns validation information about a URL.
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 *
+	 * @return WP_REST_Response|WP_Error Response object on success, or WP_Error object on failure.
+	 */
+	public function get_validated_url( $request ) {
+		$post_id       = (int) $request['id'];
+		$validated_url = new ValidatedUrlDataProvider( $post_id );
+
+		if ( is_wp_error( $validated_url->is_valid() ) ) {
+			return $validated_url->is_valid();
+		}
+
+		$data = [
+			'id'          => $validated_url->get_id(),
+			'url'         => $validated_url->get_url(),
+			'date'        => $validated_url->get_date(),
+			'author'      => $validated_url->get_author(),
+			'stylesheets' => $validated_url->get_stylesheets(),
+		];
```

Comment:
It's intentional that there's no single implicit `$validated_url->get_data()` call. The reasoning is that the REST controller is the place where the response data should be defined explicitly.